armi/nuclearDataIO/cccc/dif3d.py

```diff
@@ -0,0 +1,263 @@
+# Copyright 2019 TerraPower, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Module for reading from and writing to DIF3D files, which are module dependent
+binary inputs for the dif3d code.
+"""
+
+from armi import runLog
+from armi.nuclearDataIO import cccc
+
+
+FILE_SPEC_2D_PARAMS = (
+    [
+        "IPROBT",
+        "ISOLNT",
+        "IXTRAP",
+        "MINBSZ",
+        "NOUTMX",
+        "IRSTRT",
+        "LIMTIM",
+        "NUPMAX",
+        "IOSAVE",
+        "IOMEG1",
+        "INRMAX",
+        "NUMORP",
+        "IRETRN",
+    ]
+    + [f"IEDF{e}" for e in range(1, 11)]
+    + [
+        "NOUTBQ",
+        "I0FLUX",
+        "NOEDIT",
+        "NOD3ED",
+        "ISRHED",
+        "NSN",
+        "NSWMAX",
+        "NAPRX",
+        "NAPRXZ",
+        "NFMCMX",
+        "NXYSWP",
+        "NZSWP",
+        "ISYMF",
+        "NCMRZS",
+        "ISEXTR",
+        "NPNO",
+        "NXTR",
+        "IOMEG2",
+        "IFULL",
+        "NVFLAG",
+        "ISIMPL",
+        "IWNHFL",
+        "IPERT",
+        "IHARM",
+    ]
+)
+
+FILE_SPEC_3D_PARAMS = [
+    "EPS1",
+    "EPS2",
+    "EPS3",
+    "EFFK",
+    "FISMIN",
+    "PSINRM",
+    "POWIN",
+    "SIGBAR",
+    "EFFKQ",
+    "EPSWP",
+] + [f"DUM{e}" for e in range(1, 21)]
+
+
+class Dif3dData(cccc.DataContainer):
+    def __init__(self):
+        cccc.DataContainer.__init__(self)
+
+        self.twoD = {e: None for e in FILE_SPEC_2D_PARAMS}
+        self.threeD = {e: None for e in FILE_SPEC_3D_PARAMS}
+        self.fourD = None
+        self.fiveD = None
+
+
+class Dif3dStream(cccc.StreamWithDataContainer):
+    @staticmethod
+    def _getDataContainer() -> Dif3dData:
+        return Dif3dData()
+
+    def _rwFileID(self):
+        """
```

Comment:
I guess if we use the `-> None` type hinting, we can get rid of the Returns field?